### ISSUE TITLE
Support Login with Username also

### ISSUE THREAD

**Issue by mpscholten** [digitallyinduced/ihp-forum#2]
Right now the login only works with the email address used for sign up

**Comment by Montmorency** [comment]
It seems like the login at the moment is through the `IHP.AuthSupport.Controller.Sessions` module. Rather than issue a PR there, would the idea be to define a special `createSessionAction` for the ihp-forum? 

Atm the solution I'm thinking of involves testing a few different optional fields for authentication. This would be done by using a function which accepts a list of tuples; so we would have something like: 

```
let validFields = [(#email, (param @Text "email")), (#user, param @Text "user")]
query @record |> tryAFewFields validFields
```
 
where

```
tryAFewFields x:xs q = case (findMaybeBy (fst x) (snd x) q) of 
                                            Just _ -> \validatePassword
                                            Nothing -> tryAFewFields xs 

tryAFewFields [] _ = do 
                                    setErrorMessage "Invalid Credentials"
                                    redirectTo buildNewSessionAction
```

So to summarize:

```
do 
    let validFields = [(#email, (param @Text "email")), (#user, param @Text "user")]
    query @record 
          |> tryAFewFields [(#email, (param @Text "email")), (#user, param @Text "user")]
```

This way would require a couple different fields on the view. Alternatively we could have a single text input field  named "email_or_username" and take a single `param @Text "email_or_username"`  and test against both fields?
